src/Microsoft.Identity.Client/Platforms/net45/SilentWebUI.cs

```diff
@@ -44,7 +44,7 @@ internal class SilentWebUI : WebUI, IDisposable
         /// This is how long all redirect navigations are allowed to run for before a graceful
         /// termination of the entire browser based authentication process is attempted.
         /// </summary>
-        private const int NavigationOverallTimeout = 2000;
+        private const int NavigationOverallTimeout = 10000;
```

Comment:
Agreed with this one, 2s seems like a very strict timeout given there is a network call in the middle.

Reply:
Makes sense to set the timeout to 10s